k9mail/src/main/java/com/fsck/k9/view/MessageCryptoDisplayStatus.java

```diff
@@ -237,12 +252,15 @@ private static MessageCryptoDisplayStatus getDisplayStatusForPgpResult(CryptoRes
             }
         }
 
+        final long delta = signatureResult.getSignatureTimestamp().getTime() - messageSentDate.getTime();
```

Comment:
I think we can simply always display the error in that case. I would expect messages without a date header to be very rare, so are signed messages, so the disruption this causes should be low. I also can't think of a way to handle this otherwise in a safe manner.